internal/server/http/http.go

```diff
@@ -0,0 +1,78 @@
+package http
```

Comment:
Since this file implements `Server`, consider renaming it to `server.go` (so we have `/internal/server/http/server.go`).